src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerLoadFailureEventArgs.cs

```diff
@@ -37,9 +38,11 @@ public enum FailureErrorCode
         /// </summary>
         public Exception? Exception { get; }
 
+        public Version? ReferencedCompilerVersion { get; init; }
```

Comment:
The idea here was to prevent breaking the signature of the constructor, since it is public API. I guess we could use 'internal init' here as well.

Instead of adding this property, we could consider "smuggling through" a string representation of the referenced version through the 'message' property. But that is very hacky.

Ideally the public API would be much more limited here, i.e. the constructor would be internal and perhaps the entire type as well. It's not clear to me why external users would need to dig through why analyzers are failing to load instead of just surfacing the diagnostics.

Reply:
@mavasani do you know whether/how the IDE is directly using `new AnalyzerLoadFailureEventArgs` and related APIs such as `AnalyzerLoadFailureEventArgs.FailureErrorCode`?

Reply:
The one point I would raise is whether the increase in complexity in the API is justified by how much clearer it makes the error. Basically is the error message without this good enough? Does knowing the exact version referenced by the analyzer help the customer enough to make this API worth it? Lacking this data the customer would still have the version of the compiler and the version of the analyzer (usually it's the full path to the NuPkg). 

Leaned both ways thinking about this in last 5 minutes.

Reply:
> Lacking this data the customer would still have the version of the compiler and the version of the analyzer

The user would have the *running* version of the compiler, to be clear. They could be referencing an analyzer version which just happens to require a later SDK than the one they're running, and the process of figuring out manually which compiler version the analyzer is referencing may not be something we expect users to know how to do.

I'm happy to look at any way to pass along this information in a way that's found to be more palatable. But I do think the information is worth passing along.

Reply:
> @mavasani do you know whether/how the IDE is directly using new AnalyzerLoadFailureEventArgs and related APIs such as AnalyzerLoadFailureEventArgs.FailureErrorCode?

@RikkiGibson `AnalyzerLoadFailureEventArgs` is required to be exposed for all analyzer hosts that might require loading the analyzer assembly in contexts outside analyzer execution. For example, in the IDE, we need to load the analyzer assembly to populate the `Analyzers` node in the solution explorer, and any failure to load the analyzer assembly needs to be reported as a diagnostic in the error list. We do so by hooking up to the event handler `AnalyzerFileReference.AnalyzerLoadFailed`: https://github.com/dotnet/roslyn/blob/ec6da663c592238cca8e145044e7410c4ca9213a/src/VisualStudio/Core/Def/ProjectSystem/VisualStudioAnalyzer.cs#L52-L53

In the hooked up event handler, we create the appropriate diagnostic using the failure code and report it in the error list: https://github.com/dotnet/roslyn/blob/ec6da663c592238cca8e145044e7410c4ca9213a/src/VisualStudio/Core/Def/ProjectSystem/VisualStudioAnalyzer.cs#L61-L70